### PR TITLE
ui apis for dalle-mini

### PR DESCRIPTION
**Pull request by pngwn**
Popped this up for everyone to take a look. Feeel free to propose alternative solutions, names, etc.

This was the demo i was able to create with the addition of these APIs.

Screenshot 2022-05-13 at 23 54 30

The code looks like this:

```py
with gr.Blocks(css=".container { max-width: 800px; margin: auto; }") as demo:
    gr.Markdown("<h1><center>DALL·E mini</center></h1>")
    gr.Markdown(
        "DALL·E mini is an AI model that generates images from any prompt you give!"
    )
    with gr.Group():
        with gr.Box():
            with gr.Row().style(mobile_collapse=False, equal_height=True):
                text = gr.Textbox(
                    label="Enter your prompt", show_label=False, max_lines=1
                ).style(
                    border=(True, False, True, True),
                    margin=False,
                    rounded=(True, False, False, True),
                    container=False,
                )
                btn = gr.Button("Run").style(
                    margin=False,
                    rounded=(False, True, True, False),
                )
        gallery = gr.Gallery(label="Generated images", show_label=False).style(
            grid=(1, 3),
            height="auto",
        )
    btn.click(img, inputs=text, outputs=gallery)
```

### CODE REVIEW

File: ui/packages/app/tailwind.config.js [hunk 1]
```diff
@@ -40,5 +40,100 @@ module.exports = {
...
+		"2xl:grid-cols-4",
+		"2xl:grid-cols-5",
+		"2xl:grid-cols-6",
+		"2xl:grid-cols-7",
+		"2xl:grid-cols-8",
+		"2xl:grid-cols-9",
+		"2xl:grid-cols-10",
+		"2xl:grid-cols-11",
+		"2xl:grid-cols-12"
+	]
```
Comment (pngwn) [inline ui/packages/app/tailwind.config.js:138]: We'll figure something else out for this at some point.

File: ui/packages/atoms/src/Block.svelte [hunk 2]
```diff
@@ -63,6 +64,14 @@
...
+	class:!p-0={disable}
+	class:!m-0={disable}
+	class:!border-0={disable}
+	class:!shadow-none={disable}
+	class:overflow-visible={disable}
 >
 	<slot />
 </svelte:element>
+
+<!-- !rounded-br-none !rounded-bl-none !rounded-tr-none !rounded-bl-none -->
```
Comment (aliabid94) [inline ui/packages/atoms/src/Block.svelte:76]: remove?

Comment (aliabid94) [inline ui/packages/atoms/src/Block.svelte:76]: remove?

File: gradio/components.py [hunk 3]
```diff
@@ -3099,7 +3114,31 @@ def postprocess(self, y):
...
+    def style(
+        self,
+        rounded: Optional[bool | Tuple[bool, bool, bool, bool]] = None,
+        bg_color: Optional[str] = None,
+        text_color: Optional[str] = None,
+        margin: Optional[bool | Tuple[bool, bool, bool, bool]] = None,
+        grid: Optional[int] = None,
+        height: Optional[str] = None,
+    ):
+        if rounded is not None:
```
Comment (aliabid94) [inline gradio/components.py:3126]: use IOComponent.style() here?
e.g.
```
def style(
        self,
        rounded: Optional[bool | Tuple[bool, bool, bool, bool]] = None,
        bg_color: Optional[str] = None,
        text_color: Optional[str] = None,
        margin: Optional[bool | Tuple[bool, bool, bool, bool]] = None,
        grid: Optional[int] = None,
        height: Optional[str] = None,
    ):
        if grid is not None:
            self._style["grid"] = grid
        return IOComponent.style(
            self,
            rounded=rounded,
            bg_color=bg_color,
            text_color=text_color,
            margin=margin,
            height=height,
        )

**Review by aliabid94 — APPROVED** [review]
Small suggestions, otherwise looks great!

**Comment by abidlabs** [comment]
Love it!

**Comment by abidlabs** [comment]
Will merge it in so that I can start working on all of the merge conflicts that are bound to happen soon

**Pull request merged**